app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+class PasswordResetsController < ApplicationController
+  before_action :setup_user, :valid_user, :check_expiration, only: [:edit, :update]
+
+  def new
+  end
+
+  def edit
+  end
+
+  def create
+    @user = User.find_by(email: params[:password_reset][:email].downcase)
+    if @user
+      @user.create_reset_digest
+      UserMailer.password_reset(@user).deliver_now
```

Comment:
Well, I understand why you use synchronous `deliver_now` method — but generally you'd definitely want to use background jobs here — especially if your outgoing SMTP is not very reliable, and you need retransmissions.